swatch-core/src/main/java/org/candlepin/subscriptions/db/SubscriptionRepository.java

```diff
@@ -180,10 +178,11 @@ private static Specification<Subscription> hasBillingProviderId() {
             builder.notEqual(root.get(Subscription_.billingProviderId), ""));
   }
 
-  private static Specification<Subscription> productNameIn(Set<String> productNames) {
+  private static Specification<Subscription> productTagEquals(String productTag) {
     return (root, query, builder) -> {
-      var offeringRoot = root.get(Subscription_.offering);
-      return offeringRoot.get(Offering_.productName).in(productNames);
+      var offeringJoin = root.join(Subscription_.offering);
```

Comment:
As we discussed this is creating second offering join in the query so we have to think of another way

Reply:
@kartikshahc I revised this to use isMember() which creates a query like this and no duplicate offering join:
```

select
s1_0.start_date,s1_0.subscription_id,s1_0.billing_account_id,s1_0.billing_provider,s1_0.billing_provider_id,s1_0.end_date,o1_0.sku,o1_0.cores,o1_0.derived_sku,o1_0.description,o1_0.has_unlimited_usage,o1_0.hypervisor_cores,o1_0.hypervisor_sockets,o1_0.metered,o1_0.product_family,o1_0.product_name,o1_0.role,o1_0.sla,o1_0.sockets,o1_0.usage,s1_0.org_id,s1_0.quantity,sm1_0.start_date,sm1_0.subscription_id,sm1_0.measurement_type,sm1_0.metric_id,sm1_0.value,s1_0.subscription_number
from subscription s1_0 join offering o1_0 on o1_0.sku=s1_0.sku left join
subscription_measurements sm1_0 on s1_0.start_date=sm1_0.start_date and
s1_0.subscription_id=sm1_0.subscription_id where 1=1 and ? in
(select pt1_0.product_tag from sku_product_tag pt1_0 where
o1_0.sku=pt1_0.sku) order by s1_0.subscription_id,s1_0.start_date
```